### PR TITLE
Add Billing account to deployer

### PR DESCRIPTION
**Pull request by mjudeikis**
This retrofits billing account into deployer codebase

### CODE REVIEW

**Comment by mjudeikis** [comment]
Still missing changes to VSTS repos and pipelines.

File: pkg/deploy/generator/resources.go [hunk 1]
```diff
@@ -1554,3 +1555,64 @@ func (g *generator) acrRbac() []*arm.Resource {
...
+		Resource: &mgmtstorage.Account{
+			Sku: &mgmtstorage.Sku{
+				Name: "Standard_LRS",
+				Kind: mgmtstorage.StorageV2,
+			},
+			Name:     to.StringPtr("[parameters('billingStorageAccountName')]"),
+			Location: to.StringPtr("[resourceGroup().location]"),
+			Type:     to.StringPtr("Microsoft.Storage/storageAccounts"),
+			AccountProperties: &mgmtstorage.AccountProperties{
+				AccessTier:             mgmtstorage.Cool,
```
Comment (jim-minter) [inline pkg/deploy/generator/resources.go:1570]: remove?

File: pkg/deploy/generator/resources.go [hunk 2]
```diff
@@ -1554,3 +1555,64 @@ func (g *generator) acrRbac() []*arm.Resource {
 		},
 	}
 }
+
+func (g *generator) billingStorageAccount() *arm.Resource {
+	return &arm.Resource{
+		Resource: &mgmtstorage.Account{
+			Sku: &mgmtstorage.Sku{
+				Name: "Standard_LRS",
+				Kind: mgmtstorage.StorageV2,
```
Comment (jim-minter) [inline pkg/deploy/generator/resources.go:1564]: remove?

File: pkg/deploy/generator/resources.go [hunk 3]
```diff
@@ -1554,3 +1555,64 @@ func (g *generator) acrRbac() []*arm.Resource {
...
+			},
+			Name:     to.StringPtr("[parameters('billingStorageAccountName')]"),
+			Location: to.StringPtr("[resourceGroup().location]"),
+			Type:     to.StringPtr("Microsoft.Storage/storageAccounts"),
+			AccountProperties: &mgmtstorage.AccountProperties{
+				AccessTier:             mgmtstorage.Cool,
+				EnableHTTPSTrafficOnly: to.BoolPtr(true),
```
Comment (jim-minter) [inline pkg/deploy/generator/resources.go:1571]: remove? it's the default

File: pkg/deploy/generator/resources.go [hunk 4]
```diff
@@ -1554,3 +1555,64 @@ func (g *generator) acrRbac() []*arm.Resource {
...
+			},
+		},
+		APIVersion: azureclient.APIVersions["Microsoft.Storage"],
+	}
+}
+
+func (g *generator) roleDefinitionBillingStorageAccount() *arm.Resource {
```
Comment (jim-minter) [inline pkg/deploy/generator/resources.go:1578]: can you use the built-in `Reader and Data Access` role instead of this?

**Comment by jim-minter** [comment]
Please prepare an equivalent PR against the RP-Config repo.

**Comment by mjudeikis** [comment]
@jim-minter: PR in VSTS 2988863

**Comment by ehashman** [comment]
@mjudeikis @jim-minter what's the status of this PR?

**Comment by ehashman** [comment]
Please reopen when this is ready for review. Doing a little housekeeping

**Pull request closed without merging**